File: flutter_app/lib/features/productos/domain/usecases/create_producto.dart
import '../../../../core/usecase/usecase.dart';
import '../entities/producto.dart';
import '../repositories/producto_repository.dart';

/// Crea un producto (POST a la API) y lo agrega al estado en memoria.
class CreateProducto implements UseCase<Producto, Producto> {
  final ProductoRepository repository;
  CreateProducto(this.repository);

  @override
  Future<Producto> call(Producto producto) => repository.createProducto(producto);
}

File: react_native_app/src/features/productos/domain/usecases/createProducto.ts
import { UseCase } from '../../../../core/usecase/usecase';
import { Producto } from '../entities/Producto';
import { ProductoRepository } from '../repositories/ProductoRepository';

/** Crea un producto (POST a la API) y lo agrega al estado en memoria. */
export class CreateProducto implements UseCase<Producto, Producto> {
  constructor(private readonly repository: ProductoRepository) {}

  call(producto: Producto): Promise<Producto> {
    return this.repository.createProducto(producto);
  }
}
